GEODE-3969: Remove CallbackArgs from protobuf requests

WireBaron opened the pull request:
@galen-pivotal @kohlmu-pivotal @bschuchardt @PivotalSarge 

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [x] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [x] Is your initial contribution a single, squashed commit?

- [x] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

galen-pivotal commented:
It looks like this is actually two changes: removing callbackArgs and renaming ServerAPI to LocatorAPI. Looks good though.

upthewaterspout commented on geode-protobuf/src/main/proto/v1/clientProtocol.proto:
```diff
@@ -35,7 +35,6 @@ message Message {
 message Request {
-    CallbackArguments callbackArg = 1;
     oneof requestAPI {
         PutRequest putRequest = 2;
```
Since callback arg is gone, should putRequest actually start with 1?

galen-pivotal replied:
I kind of want to keep the one available just in case, but I don't think it's actually a good idea. Probably we should decrement all the indices.

WireBaron replied:
I decided to address this by changing the requestAPI field numbers to start at 10.  This lets us add some fields to these messages in the future without having to worry about colliding with the requestAPI field numbers (as we do expect to keep adding new types of messages here).

WireBaron commented:
Changed the fields numbers in clientProtocol.proto and removed the server_api.proto rename (it now has it's own PR).  Please take another look and push if satisfactory.

galen-pivotal commented:
@WireBaron It doesn't look like you've pushed the changes where field numbers start at 10.

WireBaron commented:
Good catch, thanks Galen.  I apparently didn't have that file checked in when I did my last commit.
